x/mint/module.go

```diff
@@ -236,9 +236,11 @@ func provideModuleBasic() runtime.AppModuleBasicWrapper {
 type mintInputs struct {
 	depinject.In
 
-	Config *modulev1.Module
-	Key    *store.KVStoreKey
-	Cdc    codec.Codec
+	Config                 *modulev1.Module
+	Key                    *store.KVStoreKey
+	Cdc                    codec.Codec
+	Authority              types.MintAuthority          `optional:"true"`
```

Comment:
slightly lost where this gets set? I thought you would need to modify protobuf config of each module?

Reply:
We set it in the app.go in `depinject.Supply` (see comments). This is because it could be just an address or we want to get a module address, so I don't think it can be in the app_config / app.yaml.

Reply:
makes sense. still trying to wrap my head around some of this.